### ISSUE TITLE
Cross-compile for scala 3?

### ISSUE THREAD

**Issue by Daenyth** [spray/spray-json#341]
I saw another issue suggesting to reuse the 2.13 artifacts, but that's very undesirable for other libraries which depend on spray-json - it's not a good approach to compatibility. For example, https://github.com/scala-js/scala-js-dom/issues/451

**Comment by raboof** [comment]
I agree, for 'direct' dependencies the cross-dependency approach is great, but for libraries it is more painful (it is for example also discouraged in https://scala-lang.org/blog/2021/04/08/scala-3-in-sbt.html#a-smooth-transition).

I made a quick attempt in #342 but quickly ran into https://github.com/lampepfl/dotty/issues/2214
